### PR TITLE
Fix protobuf-c v1.4.0 breaking CRIU's build

### PR DESCRIPTION
**Pull request by ZeyadYasser**
Fixes: #1560

The latest protobuf-c compiler breaks CRIU because they removed
leading underscores from structs in 1.4.0.

This replaces those definitions with the standard generated structs.

Signed-off-by: Zeyad Yasser <zeyady98@gmail.com>

### CODE REVIEW

**Comment by rst0git** [comment]
@ZeyadYasser it looks like Fedora, Ubuntu, Debian, and Alpine are still using protobuf-c 1.3.X.
Could you please add the Arch linux test from https://github.com/checkpoint-restore/criu/pull/1225/commits/36ca88e434168c49415e54509bb42e4020b4b07f to be able to run the tests with protobuf-c 1.4.X?

File: scripts/build/Dockerfile.archlinux [hunk 1]
```diff
@@ -0,0 +1,43 @@
+FROM archlinux
+ARG CC=gcc
+ARG ENV1=FOOBAR
+
+RUN pacman -Syu --noconfirm \
+	$CC \
+	bash \
+	make \
+	ccache \
```
Comment (rst0git) [inline scripts/build/Dockerfile.archlinux:9]: The ccache setup was removed in https://github.com/checkpoint-restore/criu/pull/1395

**Comment by ZeyadYasser** [comment]
I am not sure why the tests are compiled but not run in archlinux CI.
https://github.com/checkpoint-restore/criu/runs/3133283431?check_suite_focus=true

File: .github/workflows/archlinux-test.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run Arch Linux Test
+      run: sudo make -C scripts/ci archlinux
```
Comment (rst0git) [inline .github/workflows/archlinux-test.yml:11]: You might need to update [scripts/ci/Makefile](https://github.com/checkpoint-restore/criu/blob/47611e668e49fe0b17c39d9b2b2d83b04b2b4cb6/scripts/ci/Makefile) as well.

**Comment by ZeyadYasser** [comment]
For archlinux test, ```$(GITHUB_ACTIONS)``` is not detected
https://github.com/checkpoint-restore/criu/blob/47611e668e49fe0b17c39d9b2b2d83b04b2b4cb6/scripts/ci/Makefile#L30-L36

```log
2021-07-22T11:41:11.8447010Z docker run --env-file docker.env --rm -it --privileged -v /lib/modules:/lib/modules --tmpfs /run criu-archlinux scripts/ci/run-ci-tests.sh
2021-07-22T11:41:11.8885622Z the input device is not a TTY
2021-07-22T11:41:11.8921481Z make: *** [Makefile:59: archlinux] Error 1
```

File: .github/workflows/archlinux-test.yml (hunk 2 above)
Comment (ZeyadYasser) [reply .github/workflows/archlinux-test.yml:11]: Thanks I missed that. My bad :)

**Comment by adrianreber** [comment]
> For archlinux test, `$(GITHUB_ACTIONS)` is not detected
> 
> https://github.com/checkpoint-restore/criu/blob/47611e668e49fe0b17c39d9b2b2d83b04b2b4cb6/scripts/ci/Makefile#L30-L36
> 
> ```
> 2021-07-22T11:41:11.8447010Z docker run --env-file docker.env --rm -it --privileged -v /lib/modules:/lib/modules --tmpfs /run criu-archlinux scripts/ci/run-ci-tests.sh
> 2021-07-22T11:41:11.8885622Z the input device is not a TTY
> 2021-07-22T11:41:11.8921481Z make: *** [Makefile:59: archlinux] Error 1
> ```

You need to use `sudo -E` to preserve environment variables through sudo.

File: scripts/build/Dockerfile.archlinux [hunk 3]
```diff
@@ -0,0 +1,45 @@
+FROM archlinux
```
Comment (adrianreber) [inline scripts/build/Dockerfile.archlinux:1]: Please do not use just a shortname. Use something like `docker.io/library/archlinux:latest
`

File: scripts/build/Dockerfile.archlinux [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ARG CC=gcc
+ARG ENV1=FOOBAR
```
Comment (adrianreber) [inline scripts/build/Dockerfile.archlinux:3]: Probably unnecessary.

File: scripts/build/Dockerfile.archlinux [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	python-pip \
+	python-protobuf \
+	which \
+	sudo
+
+COPY . /criu
+WORKDIR /criu
+RUN make mrproper && date && make -j $(nproc) CC="$CC" && date
+
+RUN pacman -Syu --noconfirm \
```
Comment (adrianreber) [inline scripts/build/Dockerfile.archlinux:29]: Can you install all packages in one run of pacman?

File: scripts/build/Dockerfile.archlinux [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	bash \
+	go \
+	python-yaml \
+	flake8 \
+	asciidoctor
+
+# The rpc test cases are running as user #1000, let's add the user
+RUN useradd -u 1000 test
+
+RUN pip3 install junit_xml
```
Comment (adrianreber) [inline scripts/build/Dockerfile.archlinux:43]: Isn't there an arch package for this?

File: scripts/build/Dockerfile.archlinux [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+RUN make -C test/zdtm
```
Comment (adrianreber) [inline scripts/build/Dockerfile.archlinux:45]: New line at end of file missing.

**Review by adrianreber — CHANGES_REQUESTED** [review]

File: scripts/build/Dockerfile.archlinux [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+	iptables \
+	nftables \
+	iproute2 \
+	tar \
+	bash \
+	go \
+	python-yaml \
+	flake8 \
+	asciidoctor \
+	python-junit-xml
```
Comment (rst0git) [inline scripts/build/Dockerfile.archlinux:33]: > ####### Test zdtm/static/bridge FAIL at Deps check /usr/bin/diff failed ########

The [diffutils](https://archlinux.org/packages/core/x86_64/diffutils/) package should provide `/usr/bin/diff`

**Comment by adrianreber** [comment]
@ZeyadYasser I am myself also not sure how unresolved conversation are to be handled correctly, but I see you made the changes and do not mark them as resolved. I usually mark them as resolved once the requested changes have been done.

I am also not sure how others think to handle unresolved conversation, but from my point of view you can close them once you think they are resolved.

**Review by adrianreber — APPROVED** [review]

**Review by rst0git — APPROVED** [review]

File: test/zdtm/static/netns-nf.desc [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {   'deps': [   '/bin/sh',
                 '/sbin/iptables|/usr/sbin/iptables',
-                '/usr/lib64/xtables/libxt_standard.so|/usr/lib/iptables/libxt_standard.so|/lib/xtables/libxt_standard.so|/usr/lib/powerpc64le-linux-gnu/xtables/libxt_standard.so|/usr/lib/x86_64-linux-gnu/xtables/libxt_standard.so|/usr/lib/s390x-linux-gnu/xtables/libxt_standard.so|/usr/lib/xtables/libxt_standard.so|/usr/lib/aarch64-linux-gnu/xtables/libxt_standard.so',
+                '/lib/xtables/libxt_standard.so|/usr/lib64/xtables/libxt_standard.so|/usr/lib/iptables/libxt_standard.so|/usr/lib/powerpc64le-linux-gnu/xtables/libxt_standard.so|/usr/lib/x86_64-linux-gnu/xtables/libxt_standard.so|/usr/lib/s390x-linux-gnu/xtables/libxt_standard.so|/usr/lib/xtables/libxt_standard.so|/usr/lib/aarch64-linux-gnu/xtables/libxt_standard.so',
```
Comment (rst0git) [inline test/zdtm/static/netns-nf.desc:3]: This change should be in a separate commit.

**Review by rst0git — APPROVED** [review]
